### PR TITLE
[tracing] Add tracing for Garbage Collection

### PR DESCRIPTION
**Pull request by shreyan-gupta**
Adding logs for generic GC.
Tracing for GC during the last block of the epoch during which resharding happened.

### CODE REVIEW

File: chain/chain/src/store.rs [hunk 1]
```diff
@@ -2344,6 +2344,7 @@ impl<'a> ChainStoreUpdate<'a> {
         // Now we can proceed to removing the trie state and flat state
         let mut store_update = self.store().store_update();
         for shard_uid in prev_shard_layout.get_shard_uids() {
+            tracing::info!("GC resharding, block_hash {}, shard_uid {}", block_hash, shard_uid);
```
Comment (nagisa) [inline chain/chain/src/store.rs:2347]: Please check out the style guidelines, on tracing use at https://github.com/near/nearcore/blob/master/docs/practices/style.md#Tracing.

```suggestion
            tracing::info!(target: "...", block_hash, shard_uid, "GC resharding");
```

**Review by nagisa — CHANGES_REQUESTED** [review]

File: chain/chain/src/store.rs (hunk 1 above)
Comment (shreyan-gupta) [reply chain/chain/src/store.rs:2347]: Apologies

**Review by nagisa — APPROVED** [review]

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -996,7 +996,7 @@ impl Chain {
         tries: ShardTries,
         gc_config: &near_chain_configs::GCConfig,
     ) -> Result<(), Error> {
-        let _span = tracing::debug_span!(target: "chain", "clear_data").entered();
+        let _span = tracing::debug_span!(target: "garbage_collection", "clear_data").entered();
```
Comment (wacban) [inline chain/chain/src/chain.rs:999]: Changing the target here seems a bit controversial, what is your goal here? 
Personally I would rather keep it as chain but add the GC prefix as you've done elsewhere. Not a blocker, more of a preference and I don't think we have any guidelines on this so feel free to keep as is.

File: chain/chain/src/store.rs [hunk 3]
```diff
@@ -74,6 +74,16 @@ pub enum GCMode {
     StateSync { clear_block_info: bool },
 }
 
+impl fmt::Debug for GCMode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            GCMode::Fork(_) => write!(f, "GCMode::Fork"),
+            GCMode::Canonical(_) => write!(f, "GCMode::Canonical"),
+            GCMode::StateSync { .. } => write!(f, "GCMode::StateSync"),
```
Comment (wacban) [inline chain/chain/src/store.rs:82]: nit: You can use the f.debug_struct method. Under the hood it'll also use write but you're guaranteed that its style will be consistent with the rest of the codebase and the defaults. You can find plenty of examples in the codebase.

**Review by wacban — APPROVED** [review]

**Pull request merged**